cdap-app-fabric/src/main/java/co/cask/cdap/internal/profile/ProfileService.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.profile;
+
+import co.cask.cdap.api.Transactional;
+import co.cask.cdap.api.Transactionals;
+import co.cask.cdap.api.data.DatasetContext;
+import co.cask.cdap.common.ConflictException;
+import co.cask.cdap.common.NotFoundException;
+import co.cask.cdap.config.Config;
+import co.cask.cdap.config.ConfigDataset;
+import co.cask.cdap.config.PreferencesStore;
+import co.cask.cdap.data.dataset.SystemDatasetInstantiator;
+import co.cask.cdap.data2.dataset2.DatasetFramework;
+import co.cask.cdap.data2.dataset2.MultiThreadDatasetCache;
+import co.cask.cdap.data2.transaction.Transactions;
+import co.cask.cdap.internal.app.runtime.SystemArguments;
+import co.cask.cdap.internal.app.store.profile.ProfileDataset;
+import co.cask.cdap.proto.id.NamespaceId;
+import co.cask.cdap.proto.id.ProfileId;
+import co.cask.cdap.runtime.spi.profile.ProfileStatus;
+import com.google.common.base.Throwables;
+import org.apache.tephra.RetryStrategies;
+import org.apache.tephra.TransactionSystemClient;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+import javax.inject.Inject;
+
+/**
+ * This class is to manage profile related functions. For any store related operations,
+ * it will directly get any underlying table.
+ */
+public class ProfileService {
+  private final DatasetFramework datasetFramework;
+  private final Transactional transactional;
+
+  @Inject
+  public ProfileService(DatasetFramework datasetFramework, TransactionSystemClient txClient) {
+    this.datasetFramework = datasetFramework;
+    this.transactional = Transactions.createTransactionalWithRetry(
+      Transactions.createTransactional(new MultiThreadDatasetCache(new SystemDatasetInstantiator(datasetFramework),
+        txClient, NamespaceId.SYSTEM,
+        Collections.emptyMap(), null, null)),
+      RetryStrategies.retryOnConflict(20, 100)
+    );
+  }
+
+  /**
+   * Validate the profile status is enabled and set the preferences in same transaction
+   */
+  public void validateAndSetPreferences(String namespace, Config config) throws NotFoundException, ConflictException {
```

Comment:
this fits better in the PreferencesStore. ProfileService should have all the methods currently in ProfileStore.

Reply:
Moved to PreferenceStore